packages/workspace/src/browser/workspace-service.ts

```diff
@@ -153,21 +404,26 @@ export class WorkspaceService implements FrontendApplicationContribution {
         }
     }
 
-    protected openWindow(uri: URI, options?: WorkspaceInput): void {
+    protected async openWindow(options?: WorkspaceInput) {
```

Comment:
please always declare a return type of public API explicitly, it is a very common issue that changing a method body can change a return type of a method and break clients silently

please review all new and changed methods in this PR

Reply:
thank you for sharing ! done.